pyomo/repn/plugins/gams_writer.py

```diff
@@ -508,6 +494,47 @@ def _write_model(self,
                 output_file.write("\nput '%s' %s /;\n" % (stat, stat))
 
 
+class Categorizer(object):
+    """Class for representing categorized variables.
+
+    Given a list of variable names and a symbol map, categorizes the variable
+    names into the categories: binary, ints, positive and reals.
+
+    """
+
+    def __init__(self, var_list, symbol_map):
+        self.binary = []
+        self.ints = []
+        self.positive = []
+        self.reals = []
+
+        # categorize variables
+        for var in var_list:
+            v = symbol_map.getObject(var)
+            if v.is_binary():
+                self.binary.append(var)
+            elif v.is_integer():
+                if (v.has_lb() and (value(v.lb) >= 0)) and \
+                   (v.has_ub() and (value(v.ub) <= 1)):
+                    self.binary.append(var)
+                else:
+                    self.ints.append(var)
+            elif value(v.lb) == 0:
```

Comment:
Greater than or equal to?

Reply:
Currently, we use the positive classification to set those variables' lower bound to 0. As in we do not check (on the new line 358) for a lower bound to write for positive variables. If the lower bound is not exactly 0 they are declared as free and have a lower bound checked for and written.